src/__tests__/SSEConnection-test.js

```diff
@@ -0,0 +1,134 @@
+import { sources } from "@dschmidt/eventsourcemock";
+import SSEConnection from "../SSEConnection";
+import ConnectionEvent from "../ConnectionEvent";
+
+const url = "https://mesosphere.com";
+
+describe("Server Sent Event Connection", () => {
+  let connection;
+  beforeEach(() => {
+    connection = new SSEConnection(url, { eventTypes: ["myType"] });
+  });
+
+  describe("#constructor", () => {
+    it("fails without an url", () => {
+      expect(() => {
+        new SSEConnection();
+      }).toThrowError();
+    });
+  });
+
+  describe("#open", () => {
+    it("opens a connection", () => {
+      const onOpenEventMock = jest.fn();
+      expect(connection.state).toBe(SSEConnection.INIT);
+      connection.addListener(ConnectionEvent.OPEN, onOpenEventMock);
+
+      connection.open();
+      sources[url].emitOpen();
+
+      expect(sources[url].readyState).toBe(1);
+      expect(connection.state).toBe(SSEConnection.OPEN);
+      expect(onOpenEventMock).toHaveBeenCalled();
+    });
```

Comment:
I would actually prefer to have multiple tests for this. since the title of 'opens a connection' is not fitting. With all of the expects.

Reply:
same here, but IMHO its okay to keep this as is, maybe we want to talk about the way we're writing tests as a team at some point? feel free to block if this is bad enough for you, otherwise I'd appreciate an approval :)

Reply:
We are not doing it only here. So I am okay with it. I am just mentioning that I am not a huge fan of this 😄